numbat-cli/src/main.rs

```diff
@@ -134,13 +157,25 @@ impl Cli {
         Self {
             context: Arc::new(Mutex::new(context)),
             args,
+            config: Config::default(),
         }
     }
 
     fn run(&mut self) -> Result<()> {
         let load_prelude = !self.args.no_prelude;
         let load_init = !(self.args.no_prelude || self.args.no_init);
 
+        let user_config_path = Self::get_config_path().join("config.toml");
+
+        if let Ok(contents) = fs::read_to_string(&user_config_path) {
+            match toml::from_str(&contents) {
+                Ok(config) => self.config = config,
+                Err(_) => {
+                    bail!("Failed to parse config file")
+                }
+            }
```

Comment:
I think you can show the inner TOML parsing error here with something like
```suggestion
            let config = toml::from_str(&contents).context("Error while parsing config file")?;
```